reconciler/reconciler.go

```diff
@@ -106,15 +106,19 @@ func (r *Reconciler) wrappedInactiveEnqueue(
 	}
 }
 
-// addToqueueMap adds a *types.AccountCurrency
+// addToQueueMap adds a *types.AccountCurrency
 // to the prune map at the provided index.
-func (r *Reconciler) addToqueueMap(
+func (r *Reconciler) addToQueueMap(
 	acctCurrency *types.AccountCurrency,
 	index int64,
+	holdsLock bool,
```

Comment:
I don't hold this opinion too strongly, but instead of having this function conditionally acquire the lock, I would prefer to totally remove the lock acquisition from this function, and instead have the caller acquire/release the lock